src/darwin/Framework/CHIP/MTROTAUnsolicitedBDXMessageHandler.mm

```diff
@@ -0,0 +1,100 @@
+ */
+
+#import "MTROTAUnsolicitedBDXMessageHandler.h"
+#import "MTROTAImageTransferHandler.h"
+
+#include <protocols/bdx/BdxMessages.h>
+
+using namespace chip;
+using namespace chip::Messaging;
+using namespace chip::bdx;
+
+CHIP_ERROR MTROTAUnsolicitedBDXMessageHandler::Init(ExchangeManager * exchangeManager)
+{
+    assertChipStackLockedByCurrentThread();
+
+    VerifyOrReturnError(exchangeManager != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    MTROTAUnsolicitedBDXMessageHandler::mNumberOfDelegates = 0;
+    mExchangeMgr = exchangeManager;
+    return mExchangeMgr->RegisterUnsolicitedMessageHandlerForProtocol(Protocols::BDX::Id, this);
+}
+
+void MTROTAUnsolicitedBDXMessageHandler::Shutdown()
+{
+    assertChipStackLockedByCurrentThread();
+
+    MTROTAUnsolicitedBDXMessageHandler::mNumberOfDelegates = 0;
+    VerifyOrReturn(mExchangeMgr != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    
+    mExchangeMgr->UnregisterUnsolicitedMessageHandlerForProtocol(Protocols::BDX::Id);
+}
+
+void MTROTAUnsolicitedBDXMessageHandler::ControllerShuttingDown(MTRDeviceController * controller)
+{
+    assertChipStackLockedByCurrentThread();
+
+    // Since the OTA provider delegate only supports one BDX transfer at a time, calling ShutDown is fine for now.
+    // TODO: This class needs to keep a list of all MTROTAImageTransferHandlers mapped to the fabric index and only
+    // delete the MTROTAImageTransferHandler with a fabric index matching the MTRDeviceController's fabric index.
+    Shutdown();
+}
+
+CHIP_ERROR MTROTAUnsolicitedBDXMessageHandler::OnUnsolicitedMessageReceived(const PayloadHeader & payloadHeader, const SessionHandle & session,
+    ExchangeDelegate *  _Nonnull & newDelegate)
+{
+    assertChipStackLockedByCurrentThread();
+
+    ChipLogDetail(BDX, "MTROTAUnsolicitedBDXMessageHandler: OnUnsolicitedMessageReceived: message " ChipLogFormatMessageType " protocol " ChipLogFormatProtocolId,
+        payloadHeader.GetMessageType(), ChipLogValueProtocolId(payloadHeader.GetProtocolID()));
+
+    VerifyOrReturnError(mExchangeMgr != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    if (GetNumberOfDelegates() >= 1)
+    {
+        return CHIP_ERROR_BUSY;
+    }
+
+    // Only proceed if there is a valid fabric index for the SessionHandle.
+    if (session->IsSecureSession() && session->AsSecureSession() != nullptr && session->AsSecureSession()->GetFabricIndex() != kUndefinedFabricIndex)
+    {
+        // If we receive a ReceiveInit BDX message, create a new MTROTAImageTransferHandler and register it
+        // as the handler for all BDX messages that will come over this exchange and increment the number of delegates.
+        if (payloadHeader.HasMessageType(MessageType::ReceiveInit)) {
+            MTROTAImageTransferHandler * otaImageTransferHandler = new MTROTAImageTransferHandler();
+            newDelegate = otaImageTransferHandler;
+            return CHIP_NO_ERROR;
+        }
+    }
+    return CHIP_ERROR_INCORRECT_STATE;
```

Comment:
Returning CHIP_ERROR_INCORRECT_STATE. I looked at the spec and there is Unexpected Message status code that should be sent for any BDX messages but we have not yet established a session yet. the unexpected message does map to CHIP_ERROR_INCORRECT_STATE so thats what I used here.